src/Native/System.Native/pal_process.cpp

```diff
@@ -142,3 +142,92 @@ int32_t ForkAndExecProcess(
+// to convert our standard types into the platform specific ones.
+static
+int32_t ConvertRLimitResourcesPalToPlatform(RLimitResources value)
+{
+    switch (value)
+    {
+        case RLimitResources::PAL_RLIMIT_CPU:
+            return RLIMIT_CPU;
+        case RLimitResources::PAL_RLIMIT_FSIZE:
+            return RLIMIT_FSIZE;
+        case RLimitResources::PAL_RLIMIT_DATA:
+            return RLIMIT_DATA;
+        case RLimitResources::PAL_RLIMIT_STACK:
+            return RLIMIT_STACK;
+        case RLimitResources::PAL_RLIMIT_CORE:
+            return RLIMIT_CORE;
+        case RLimitResources::PAL_RLIMIT_AS:
+            return RLIMIT_AS;
+        case RLimitResources::PAL_RLIMIT_RSS:
+            return RLIMIT_RSS;
+        case RLimitResources::PAL_RLIMIT_MEMLOCK:
+            return RLIMIT_MEMLOCK;
+        case RLimitResources::PAL_RLIMIT_NPROC:
+            return RLIMIT_NPROC;
+        case RLimitResources::PAL_RLIMIT_NOFILE:
+            return RLIMIT_NOFILE;
+    }
+
+    assert(!"Unknown RLIMIT value");
+    return -1;
+}
+
+// Because RLIM_INFINITY is different per-platform, use the max value of a uint64 (which is RLIM_INFINITY on Ubuntu)
+// to signify RLIM_INIFINITY; on OS X, where RLIM_INFINITY is slightly lower, we'll translate it to the correct value here.
+static
+uint64_t ConvertFromManagedRLimitInfinityToPalIfNecessary(uint64_t value)
+{
+    if (value == ((uint64_t)-1))
+        return RLIM_INFINITY;
+    else
+        return value;
+}
+
+// Because RLIM_INFINITY is different per-platform, use the max value of a uint64 (which is RLIM_INFINITY on Ubuntu)
+// to signify RLIM_INIFINITY; on OS X, where RLIM_INFINITY is slightly lower, we'll translate it to the correct value here.
+static
+uint64_t ConvertFromNativeRLimitInfinityToManagedIfNecessary(uint64_t value)
+{
+    if (value == RLIM_INFINITY)
+        return ((uint64_t)-1);
+    else
+        return value;
+}
+
+extern "C"
+int32_t GetRLimit(
+    RLimitResources     resourceType,
+    RLimit*             limits)
+{
+    assert(limits != nullptr);
+
+    int32_t platformLimit = ConvertRLimitResourcesPalToPlatform(resourceType);
+    rlimit internalLimit;
+    int result = getrlimit(platformLimit, &internalLimit);
+    if (result == 0)
+    {
+        *limits = { ConvertFromNativeRLimitInfinityToManagedIfNecessary(internalLimit.rlim_cur), ConvertFromNativeRLimitInfinityToManagedIfNecessary(internalLimit.rlim_max) };
+    }
+    else
+    {
+        *limits = { };
+    }
+    
+    return result;
+}
+
+extern "C"
+int32_t SetRLimit(
+    RLimitResources     resourceType,
+    const RLimit* const limits)
+{
+    assert(limits != nullptr);
+
+    int32_t platformLimit = ConvertRLimitResourcesPalToPlatform(resourceType);
+    rlimit internalLimit = { ConvertFromManagedRLimitInfinityToPalIfNecessary(limits->CurrentLimit), ConvertFromManagedRLimitInfinityToPalIfNecessary(limits->MaximumLimit) };
```

Comment:
Is current before maximum guaranteed by POSIX? Safer to set these by name. I think it would be more readable to have rlimit -> RLimit and RLimit -> rlimit conversion helpers, which could take care of that.